ui/nodeview_space_menu.py

```diff
@@ -466,13 +469,36 @@ def draw(self, context):
         for elem in self.draw_data:
             elem.draw(self.layout)
 
+add_node_menu = None
+
+def setup_add_menu():
+    global add_node_menu
+    datafiles = Path(bpy.utils.user_resource('DATAFILES', path='sverchok', create=True))
 
-datafiles = Path(bpy.utils.user_resource('DATAFILES', path='sverchok', create=True))
-menu_file = datafiles / 'index.yaml'
-if not menu_file.exists():
     default_menu_file = Path(__file__).parents[1] / 'index.yaml'
-    shutil.copy(default_menu_file, menu_file)
-add_node_menu = Category.from_config(yaml_parser.load(menu_file), 'All Categories', icon_name='RNA')
+
+    with sv_preferences() as prefs:
+        if prefs is None:
+            raise Exception("Internal error: Sverchok preferences are not initialized yet at the moment of loading the menu")
+        if prefs.menu_preset_usage == 'COPY':
+            default_menu_file = get_menu_preset_path(prefs.menu_preset)
+            menu_file = datafiles / 'index.yaml'
+            use_preset_copy = True
+        else:
+            menu_file = get_menu_preset_path(prefs.menu_preset)
+            use_preset_copy = False
+
+    if use_preset_copy and not menu_file.exists():
+        info(f"Applying menu preset {default_menu_file} at startup")
+        shutil.copy(default_menu_file, menu_file)
+    debug(f"Using menu preset file: {menu_file}")
+    add_node_menu = Category.from_config(yaml_parser.load(menu_file), 'All Categories', icon_name='RNA')
+
+def get_add_node_menu():
+    global add_node_menu
+    if add_node_menu is None:
+        setup_add_menu()
+    return add_node_menu
```

Comment:
I can't agree less that the function represents functionality of the module. If it returns Category instances than it can/should be a class method of the Category. The same is about setup_add_menu.

Also I don't like a laze concept here and in general - it makes things more complicated than they could be. It probably would be valid to use if initialization of the menu was really in arbitrary place in code. I assume you don't know from where the menu is called what leads to treating code as a black box / magic. From my knowledge this is not good approach.

I had a look at the places and it seems that the menu is only used in panels and operators what means that if you need the settings to be available during initialization of the menu it's possible to initialize the menu in the register function.

```py
def register():
    add_menu = Category.generate_main_menu()
    add_menu.register()
```

Reply:
I will not have objections if you rewrite it according to your suggestions :)

Reply:
I'm too tired to rewrite somebody code.